packages/bpk-component-infinite-scroll/stories.js

```diff
@@ -93,11 +105,43 @@ storiesOf('bpk-component-infinite-scroll', module)
       });
     return (
       <InfiniteList
-        onItemsFetch={delayed}
+        dataSource={new MyDS(delayed)}
         elementsPerScroll={10}
         renderLoadingComponent={() => (
           <BpkSpinner type={SPINNER_TYPES.primary} />
         )}
       />
     );
+  })
+  .add('Force update data', () => {
+    class StoryDataSource extends DataSource {
+      constructor(elements) {
+        super();
+        this.fetchItems = onItemsFetch(elements);
+      }
+
+      updateElements(newElements) {
+        this.fetchItems = onItemsFetch(newElements);
+        this.triggerListeners();
```

Comment:
My idea here is that triggerListeners would be called with the `newElements` and so the components that have registered the lister will receive that new data and be able to rerender.